Lib/idlelib/idle_test/test_config.py

```diff
@@ -185,6 +187,382 @@ def test_save(self):
+             'FormatParagraph', 'ParenMatch', 'RstripExtension', 'ScriptBinding',
+             'ZoomHeight'])
+        eq(conf.GetExtensions(editor_only=True),
+           ['AutoComplete', 'AutoExpand', 'CallTips', 'CodeContext',
+            'FormatParagraph', 'ParenMatch', 'RstripExtension', 'ScriptBinding',
+            'ZoomHeight'])
+        eq(conf.GetExtensions(shell_only=True),
+           ['AutoComplete', 'AutoExpand', 'CallTips', 'FormatParagraph',
+            'ParenMatch', 'ZoomHeight'])
+
+        # Add user extensions
+        conf.SetOption('extensions', 'Foobar', 'enable', 'True')
+        eq(conf.GetExtensions(),
+           ['AutoComplete', 'AutoExpand', 'CallTips', 'CodeContext',
+            'FormatParagraph', 'ParenMatch', 'RstripExtension',
+            'ScriptBinding', 'ZoomHeight', 'Foobar'])  # User extensions didn't sort
+
+    def test_remove_key_bind_names(self):
+        conf = self.mock_config()
+
+        self.assertEqual(
+            conf.RemoveKeyBindNames(conf.GetSectionList('default', 'extensions')),
+            ['AutoComplete', 'AutoExpand', 'CallTips', 'CodeContext',
+             'FormatParagraph', 'ParenMatch', 'RstripExtension', 'ScriptBinding',
+             'ZoomHeight'])
+
+    def test_get_extn_name_for_event(self):
+        conf = self.mock_config()
+
+        eq = self.assertEqual
+        eq(conf.GetExtnNameForEvent('force-open-completions'), 'AutoComplete')
+        eq(conf.GetExtnNameForEvent('expand-word'), 'AutoExpand')
+        eq(conf.GetExtnNameForEvent('force-open-calltip'), 'CallTips')
+        eq(conf.GetExtnNameForEvent('zoom-height'), 'ZoomHeight')
+
+    def test_get_extension_keys(self):
+        conf = self.mock_config()
+
+        eq = self.assertEqual
+        eq(conf.GetExtensionKeys('AutoComplete'),
+           {'<<force-open-completions>>': ['<Control-Key-space>']})
+        eq(conf.GetExtensionKeys('ParenMatch'),
+           {'<<flash-paren>>': ['<Control-Key-0>']})
+
+        key = ['<Option-Key-2>'] if sys.platform == 'darwin' else ['<Alt-Key-2>']
+        eq(conf.GetExtensionKeys('ZoomHeight'), {'<<zoom-height>>': key})
+
+    def test_get_extension_bindings(self):
+        conf = self.mock_config()
+
+        self.assertEqual(conf.GetExtensionBindings('NotExists'), {})
+
+        key = ['<Option-Key-2>'] if sys.platform == 'darwin' else ['<Alt-Key-2>']
+        self.assertEqual(
+            conf.GetExtensionBindings('ZoomHeight'), {'<<zoom-height>>': key})
+
+        # Add non-configuarable bindings
+        conf.defaultCfg['extensions'].add_section('Foobar')
+        conf.defaultCfg['extensions'].add_section('Foobar_bindings')
+        conf.defaultCfg['extensions'].set('Foobar', 'enable', 'True')
+        conf.defaultCfg['extensions'].set('Foobar_bindings', 'foobar', '<Key-F>')
+        self.assertEqual(conf.GetExtensionBindings('Foobar'), {'<<foobar>>': ['<Key-F>']})
+
+    def test_get_current_keyset(self):
+        import sys
```

Comment:
This should be removed since it is imported at the top of the test, I'll leave here until @terryjreedy push his commit to GitHub (there is a conflict here)